### ISSUE TITLE
Installation Issue, can not install wafw00f

### ISSUE THREAD

**Issue by kali64amd** [EnableSecurity/wafw00f#35]
# ## hi, there could you please advise me about this error??????????

## **Successfully Installed wafw00f**

```
root@kali:~/wafw00f# python setup.py install
running install
running bdist_egg
running egg_info
creating wafw00f.egg-info
writing requirements to wafw00f.egg-info/requires.txt
writing wafw00f.egg-info/PKG-INFO
writing top-level names to wafw00f.egg-info/top_level.txt
writing dependency_links to wafw00f.egg-info/dependency_links.txt
writing manifest file 'wafw00f.egg-info/SOURCES.txt'
reading manifest file 'wafw00f.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'wafw00f.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib.linux-x86_64-2.7
creating build/lib.linux-x86_64-2.7/wafw00f
copying wafw00f/main.py -> build/lib.linux-x86_64-2.7/wafw00f
copying wafw00f/wafprio.py -> build/lib.linux-x86_64-2.7/wafw00f
copying wafw00f/__init__.py -> build/lib.linux-x86_64-2.7/wafw00f
copying wafw00f/manager.py -> build/lib.linux-x86_64-2.7/wafw00f
creating build/lib.linux-x86_64-2.7/wafw00f/lib
copying wafw00f/lib/proxy.py -> build/lib.linux-x86_64-2.7/wafw00f/lib
copying wafw00f/lib/evillib.py -> build/lib.linux-x86_64-2.7/wafw00f/lib
copying wafw00f/lib/__init__.py -> build/lib.linux-x86_64-2.7/wafw00f/lib
creating build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/powercdn.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/uspses.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/secureiis.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/dotdefender.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/hyperguard.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/ciscoacexml.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/netcontinuum.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/f5bigipasm.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/f5trafficshield.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/nsfocus.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/safedog.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/wzb360.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/denyall.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/missioncontrol.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/nevisproxy.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/incapsula.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/binarysec.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/f5bigipapm.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/webscurity.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/anquanbao.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/f5bigipltm.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/chinacache.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/ibm.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/webknight.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/cloudflare.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/netscaler.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/profense.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/f5firepass.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/airlock.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/isaserver.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/west263cdn.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/urlscan.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/modsecuritycrs.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/betterwpsecurity.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/__init__.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/modsecurity.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/barracuda.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/teros.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/imperva.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
copying wafw00f/plugins/ibmdatapower.py -> build/lib.linux-x86_64-2.7/wafw00f/plugins
creating build/lib.linux-x86_64-2.7/wafw00f/tests
copying wafw00f/tests/test_main.py -> build/lib.linux-x86_64-2.7/wafw00f/tests
copying wafw00f/tests/__init__.py -> build/lib.linux-x86_64-2.7/wafw00f/tests
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/wafw00f
creating build/bdist.linux-x86_64/egg/wafw00f/lib
copying build/lib.linux-x86_64-2.7/wafw00f/lib/proxy.py -> build/bdist.linux-x86_64/egg/wafw00f/lib
copying build/lib.linux-x86_64-2.7/wafw00f/lib/evillib.py -> build/bdist.linux-x86_64/egg/wafw00f/lib
copying build/lib.linux-x86_64-2.7/wafw00f/lib/__init__.py -> build/bdist.linux-x86_64/egg/wafw00f/lib
copying build/lib.linux-x86_64-2.7/wafw00f/main.py -> build/bdist.linux-x86_64/egg/wafw00f
creating build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/powercdn.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/uspses.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/secureiis.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/dotdefender.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/hyperguard.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/ciscoacexml.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/netcontinuum.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/f5bigipasm.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/f5trafficshield.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/nsfocus.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/safedog.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/wzb360.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/denyall.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/missioncontrol.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/nevisproxy.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/incapsula.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/binarysec.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/f5bigipapm.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/webscurity.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/anquanbao.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/f5bigipltm.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/chinacache.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/ibm.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/webknight.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/cloudflare.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/netscaler.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/profense.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/f5firepass.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/airlock.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/isaserver.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/west263cdn.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/urlscan.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/modsecuritycrs.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/betterwpsecurity.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/__init__.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/modsecurity.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/barracuda.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/teros.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/imperva.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
copying build/lib.linux-x86_64-2.7/wafw00f/plugins/ibmdatapower.py -> build/bdist.linux-x86_64/egg/wafw00f/plugins
creating build/bdist.linux-x86_64/egg/wafw00f/tests
copying build/lib.linux-x86_64-2.7/wafw00f/tests/test_main.py -> build/bdist.linux-x86_64/egg/wafw00f/tests
copying build/lib.linux-x86_64-2.7/wafw00f/tests/__init__.py -> build/bdist.linux-x86_64/egg/wafw00f/tests
copying build/lib.linux-x86_64-2.7/wafw00f/wafprio.py -> build/bdist.linux-x86_64/egg/wafw00f
copying build/lib.linux-x86_64-2.7/wafw00f/__init__.py -> build/bdist.linux-x86_64/egg/wafw00f
copying build/lib.linux-x86_64-2.7/wafw00f/manager.py -> build/bdist.linux-x86_64/egg/wafw00f
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/lib/proxy.py to proxy.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/lib/evillib.py to evillib.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/lib/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/main.py to main.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/powercdn.py to powercdn.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/uspses.py to uspses.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/secureiis.py to secureiis.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/dotdefender.py to dotdefender.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/hyperguard.py to hyperguard.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/ciscoacexml.py to ciscoacexml.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/netcontinuum.py to netcontinuum.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/f5bigipasm.py to f5bigipasm.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/f5trafficshield.py to f5trafficshield.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/nsfocus.py to nsfocus.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/safedog.py to safedog.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/wzb360.py to wzb360.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/denyall.py to denyall.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/missioncontrol.py to missioncontrol.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/nevisproxy.py to nevisproxy.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/incapsula.py to incapsula.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/binarysec.py to binarysec.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/f5bigipapm.py to f5bigipapm.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/webscurity.py to webscurity.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/anquanbao.py to anquanbao.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/f5bigipltm.py to f5bigipltm.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/chinacache.py to chinacache.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/ibm.py to ibm.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/webknight.py to webknight.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/cloudflare.py to cloudflare.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/netscaler.py to netscaler.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/profense.py to profense.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/f5firepass.py to f5firepass.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/airlock.py to airlock.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/isaserver.py to isaserver.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/west263cdn.py to west263cdn.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/urlscan.py to urlscan.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/modsecuritycrs.py to modsecuritycrs.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/betterwpsecurity.py to betterwpsecurity.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/modsecurity.py to modsecurity.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/barracuda.py to barracuda.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/teros.py to teros.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/imperva.py to imperva.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/plugins/ibmdatapower.py to ibmdatapower.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/tests/test_main.py to test_main.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/tests/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/wafprio.py to wafprio.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/__init__.py to __init__.pyc
byte-compiling build/bdist.linux-x86_64/egg/wafw00f/manager.py to manager.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/scripts-2.7
copying and adjusting wafw00f/bin/wafw00f -> build/scripts-2.7
changing mode of build/scripts-2.7/wafw00f from 644 to 755
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/wafw00f -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/wafw00f to 755
copying wafw00f.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying wafw00f.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying wafw00f.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying wafw00f.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying wafw00f.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
wafw00f.manager: module references __file__
creating dist
creating 'dist/wafw00f-0.9.4-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing wafw00f-0.9.4-py2.7.egg
removing '/usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg' (and everything under it)
creating /usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg
Extracting wafw00f-0.9.4-py2.7.egg to /usr/local/lib/python2.7/dist-packages
wafw00f 0.9.4 is already the active version in easy-install.pth
Installing wafw00f script to /usr/local/bin

Installed /usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg
Processing dependencies for wafw00f==0.9.4
Searching for pluginbase==0.3
Best match: pluginbase 0.3
Processing pluginbase-0.3-py2.7.egg
pluginbase 0.3 is already the active version in easy-install.pth

Using /usr/local/lib/python2.7/dist-packages/pluginbase-0.3-py2.7.egg
Searching for beautifulsoup4==4.4.1
Best match: beautifulsoup4 4.4.1
Processing beautifulsoup4-4.4.1-py2.7.egg
beautifulsoup4 4.4.1 is already the active version in easy-install.pth

Using /usr/local/lib/python2.7/dist-packages/beautifulsoup4-4.4.1-py2.7.egg
Finished processing dependencies for wafw00f==0.9.4
```

## **Then got Error when I try to run..............**

```
root@kali:~/wafw00f# wafw00f
Traceback (most recent call last):
  File "/usr/local/bin/wafw00f", line 4, in <module>
    __import__('pkg_resources').run_script('wafw00f==0.9.4', 'wafw00f')
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 744, in run_script
    
  File "build/bdist.linux-x86_64/egg/pkg_resources/__init__.py", line 1499, in run_script
    
  File "/usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg/EGG-INFO/scripts/wafw00f", line 4, in <module>
    from wafw00f import main
  File "/usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg/wafw00f/main.py", line 53, in <module>
    from wafw00f.lib.evillib import oururlparse, scrambledheader, waftoolsengine
  File "/usr/local/lib/python2.7/dist-packages/wafw00f-0.9.4-py2.7.egg/wafw00f/lib/evillib.py", line 14, in <module>
    from bs4 import BeautifulSoup
  File "build/bdist.linux-x86_64/egg/bs4/__init__.py", line 30, in <module>
    
  File "build/bdist.linux-x86_64/egg/bs4/builder/__init__.py", line 314, in <module>
    
  File "build/bdist.linux-x86_64/egg/bs4/builder/_html5lib.py", line 70, in <module>
AttributeError: 'module' object has no attribute '_base'
```

**Comment by plancksec** [comment]
same issue please assist, thank you.

**Comment by sandrogauci** [comment]
_Note: when submitting issues, please help me debug by specifying which OS and version you are trying to run this on. That usually means that I have to do less guess work. Additionally, code should be marked as such._

I was able to reproduce the issue on the latest version of Kali Linux; but not on the previous version. The issue appears to be related to bs4 and/or html5 libraries which, for some reason, is giving the error that you pasted. This issue also appears to affect other tools too when bs4 is updated on Kali 2017.1.

Also note that if you stick with the version that Kali 2017.1 comes with which is 0.9.3 (which is not extremely outdated); that does appear to work.

Wafw00f  on other distros (non-Kali) seems to be OK too according to my limited tests.

I am unsure how to address this issue at the moment but suggestions and code changes would definitely be appreciated.

**Comment by sandrogauci** [comment]
It seems like a BeautifulSoup import issue (importing an incompatable version of html5lib). I'm going to close - reopen if you have an update.

**Comment by plancksec** [comment]
Sandro,

Thank you for checking on this. I enjoy your work.

Larry

--
  Larry Smith
  PlanckSec Consulting LLC
  904.210.3369
  larry@plancksec.net



On Sun, Jun 4, 2017, at 03:35 AM, Sandro Gauci wrote:
> It seems like a BeautifulSoup import issue (importing an
> incompatable version of html5lib). I'm going to close - reopen if
> you have an update.> — You are receiving this because you commented. Reply to this email
> directly, view it on GitHub[1], or mute the thread[2].> 


Links:

  1. https://github.com/EnableSecurity/wafw00f/issues/35#issuecomment-306023994

**Comment by sandrogauci** [comment]
fixed #38

**Comment by plebbob** [comment]
I'm having a beautifulsoup issue also but not juse with wafw00f. I'm using parrot os 4.1
 
wafw00f 0.9.4 has requirement beautifulsoup4==4.4.1, but you'll have beautifulsoup4 4.6.0 which is incompatible.
